### PR TITLE
feat: create farm intro

### PR DESCRIPTION
**Pull request by nguyenvuces**
get date and display farm intro

CU-865d3gqqy

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/Code-Engine-Studio/ces-semantic-release/blob/main/README.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by ces-huy-dang — APPROVED** [review]

File: web/modules/custom/farm_page/farm_page.module [hunk 1]
```diff
@@ -22,3 +23,19 @@ function farm_page_theme() {
     ],
   ];
 }
+
+/**
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/farm_page.module:27]: /**
 * Implements hook_preprocess_block().
 */

File: web/modules/custom/farm_page/farm_page.module [hunk 2]
```diff
@@ -22,3 +23,19 @@ function farm_page_theme() {
...
+ * This function is called when rendering a block generated by your module. It
+ * attaches the 'farm_page/farm_page' library to the block's render array,
+ * allowing you to include CSS or JavaScript assets specific to your module's
+ * block.
+ *
+ * @param array &$variables
+ *   An associative array of variables for the block template.
+ */
+function farm_page_preprocess_block(&$variables) {
+  // Attach the library to all blocks generated by your module.
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/farm_page.module:39]: Comment should reflects what it does.

File: web/modules/custom/farm_page/src/Controller/FarmPageController.php [hunk 3]
```diff
@@ -88,15 +88,20 @@ public function myFarm($uid) {
         $store = Store::load($store_id);
         $products_array = $this->farmPageService->getFeatureProducts((int) $store_id);
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Controller/FarmPageController.php:89]: $products should be enough.

File: web/modules/custom/farm_page/src/Controller/FarmPageController.php [hunk 4]
```diff
@@ -88,15 +88,20 @@ public function myFarm($uid) {
...
         $store_images_field = $store->get('field_farm_image')->getValue();
-        $store_images_url = $this->farmPageService->getImagesUrlFromArray($store_images_field);
         $products = $this->farmPageService->getProductsInfomation($products_array);
         $is_exist = TRUE;
+        $farmData = [
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Controller/FarmPageController.php:93]: Variable naming convention.
image

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig [hunk 5]
```diff
@@ -1,8 +1,16 @@
+{% set farm_welcome = "WELCOME TO " ~ data.farm.name ~ "!" %}
 {% if is_exist == true %}
-{{ store.field_description.value }}
-	{% for image in store_images %}
-	<img src={{image}}>
-	{% endfor %}
-{% else %}
-{{ message }}
+	<div class="intro-farm">
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/templates/farm-homepage.html.twig:3]: Wrong code format.
https://app.clickup.com/610996/v/dc/jmnm-14225/jmnm-7825

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig [hunk 6]
```diff
@@ -1,8 +1,16 @@
...
+		<div class="farm-name">
+			<h3>{{farm_welcome|upper}}</h3>
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/templates/farm-homepage.html.twig:5]: Be consistent, put all properties to farm.
Eg: farm.name, farm.image, farm.description

File: web/modules/custom/farm_page/src/Controller/FarmPageController.php [hunk 7]
```diff
@@ -88,15 +88,20 @@ public function myFarm($uid) {
         $store = Store::load($store_id);
         $products_array = $this->farmPageService->getFeatureProducts((int) $store_id);
         $store_images_field = $store->get('field_farm_image')->getValue();
-        $store_images_url = $this->farmPageService->getImagesUrlFromArray($store_images_field);
         $products = $this->farmPageService->getProductsInfomation($products_array);
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Controller/FarmPageController.php:91]: $farmService

**Review by ces-dantran — CHANGES_REQUESTED** [review]

File: web/modules/custom/farm_page/farm_page.module (hunk 1 above)
Comment (nguyenvuces) [reply web/modules/custom/farm_page/farm_page.module]: this code is no used

File: web/modules/custom/farm_page/farm_page.module (hunk 2 above)
Comment (nguyenvuces) [reply web/modules/custom/farm_page/farm_page.module]: this comment is no used

File: web/modules/custom/farm_page/src/Controller/FarmPageController.php (hunk 4 above)
Comment (nguyenvuces) [reply web/modules/custom/farm_page/src/Controller/FarmPageController.php]: updated

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig (hunk 5 above)
Comment (nguyenvuces) [reply web/modules/custom/farm_page/templates/farm-homepage.html.twig]: updated

File: web/modules/custom/farm_page/src/Controller/FarmPageController.php (hunk 7 above)
Comment (nguyenvuces) [reply web/modules/custom/farm_page/src/Controller/FarmPageController.php]: updated

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig (hunk 6 above)
Comment (nguyenvuces) [reply web/modules/custom/farm_page/templates/farm-homepage.html.twig]: updated

File: web/modules/custom/farm_page/farm_page.module [hunk 8]
```diff
@@ -13,6 +13,7 @@ function farm_page_theme() {
     'farm_homepage' => [
       'template' => 'farm-homepage',
       'variables' => [
+        'data' => [],
         'store' => 'No value',
         'products' => 'Store has no product',
         'store_images' => 'No images',
```
Comment (ces-khoado) [inline web/modules/custom/farm_page/farm_page.module:19]: If we're already using data[] to store variables then we don't need these. Return them as key => value of data[] and call them by data. in template.

**Review by ces-khoado — APPROVED** [review]

**Review by ces-dantran — APPROVED** [review]

**Pull request merged**